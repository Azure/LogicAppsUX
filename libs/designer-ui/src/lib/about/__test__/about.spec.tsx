import React from 'react';
import type { AboutProps } from '../';
import { About } from '../';
import * as ReactShallowRenderer from 'react-test-renderer/shallow';
import { describe, vi, beforeEach, afterEach, beforeAll, afterAll, it, test, expect } from 'vitest';
describe('lib/monitoring/requestpanel/request', () => {
  let minimal: AboutProps, renderer: ReactShallowRenderer.ShallowRenderer;

  beforeEach(() => {
    minimal = {};
    renderer = ReactShallowRenderer.createRenderer();
  });

  afterEach(() => {
    renderer.unmount();
  });

  it('should render', () => {
    const tree = renderer.render(<About {...minimal} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render a body', () => {
    const props = {
      ...minimal,
      connectorDisplayName: 'test',
      description: 'This is a description ',
      headerIcons: [
        { title: 'Tag1', badgeText: 'test' },
        { title: 'Tag2', badgeText: 'more' },
      ],
    };

    const tree = renderer.render(<About {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render a description with external link', () => {
    const props = {
      ...minimal,
      connectorDisplayName: 'test',
      description: 'This is a description ',
      descriptionDocumentation: { url: 'www.microsoft.com', description: 'more info' },
      headerIcons: [
        { title: 'Tag1', badgeText: 'test' },
        { title: 'Tag2', badgeText: 'more' },
      ],
    };

    const tree = renderer.render(<About {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with different connectorDisplayName values', () => {
    const props = {
      ...minimal,
      connectorDisplayName: 'new test',
    };
    const tree = renderer.render(<About {...props} />);
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with different headerIcons values', () => {
    const props = {
      ...minimal,
      headerIcons: [
        { title: 'NewTag1', badgeText: 'new test' },
        { title: 'NewTag2', badgeText: 'new more' },
      ],
    };
    const tree = renderer.render(<About {...props} />);
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly when descriptionDocumentation is undefined', () => {
    const props = {
      ...minimal,
      descriptionDocumentation: undefined,
    };
    const tree = renderer.render(<About {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
