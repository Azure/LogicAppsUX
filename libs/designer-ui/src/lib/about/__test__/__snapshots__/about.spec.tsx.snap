// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`lib/monitoring/requestpanel/request > renders correctly when descriptionDocumentation is undefined 1`] = `
<div
  className="msla-panel-about-container"
>
  <div
    className="msla-panel-about-name"
  >
    <Label
      className="msla-panel-connector-label"
      size="large"
    >
      Connector
    </Label>
    <div
      className="msla-panel-connector-name"
    >
      Not available
    </div>
  </div>
  <div
    className="msla-panel-about-description"
  >
    <Label
      className="msla-panel-description-label"
      size="large"
    >
      Operation note
    </Label>
    <div
      className="msla-panel-description"
    >
      <DocumentationItem />
    </div>
  </div>
  <div
    className="msla-panel-about-tags"
  >
    <Label
      className="msla-panel-tags-label"
      size="large"
    >
      Tags
    </Label>
    <div
      className="msla-panel-tags"
    >
      None
    </div>
  </div>
</div>
`;

exports[`lib/monitoring/requestpanel/request > renders correctly with different connectorDisplayName values 1`] = `
<div
  className="msla-panel-about-container"
>
  <div
    className="msla-panel-about-name"
  >
    <Label
      className="msla-panel-connector-label"
      size="large"
    >
      Connector
    </Label>
    <div
      className="msla-panel-connector-name"
    >
      new test
    </div>
  </div>
  <div
    className="msla-panel-about-description"
  >
    <Label
      className="msla-panel-description-label"
      size="large"
    >
      Operation note
    </Label>
    <div
      className="msla-panel-description"
    >
      <DocumentationItem />
    </div>
  </div>
  <div
    className="msla-panel-about-tags"
  >
    <Label
      className="msla-panel-tags-label"
      size="large"
    >
      Tags
    </Label>
    <div
      className="msla-panel-tags"
    >
      None
    </div>
  </div>
</div>
`;

exports[`lib/monitoring/requestpanel/request > renders correctly with different headerIcons values 1`] = `
<div
  className="msla-panel-about-container"
>
  <div
    className="msla-panel-about-name"
  >
    <Label
      className="msla-panel-connector-label"
      size="large"
    >
      Connector
    </Label>
    <div
      className="msla-panel-connector-name"
    >
      Not available
    </div>
  </div>
  <div
    className="msla-panel-about-description"
  >
    <Label
      className="msla-panel-description-label"
      size="large"
    >
      Operation note
    </Label>
    <div
      className="msla-panel-description"
    >
      <DocumentationItem />
    </div>
  </div>
  <div
    className="msla-panel-about-tags"
  >
    <Label
      className="msla-panel-tags-label"
      size="large"
    >
      Tags
    </Label>
    <div
      className="msla-panel-tags"
    >
      <React.Fragment>
        <div
          aria-label="new test"
          className="msla-panel-tag"
          title="NewTag1"
        >
          new test
        </div>
        <div
          aria-label="new more"
          className="msla-panel-tag"
          title="NewTag2"
        >
          new more
        </div>
      </React.Fragment>
    </div>
  </div>
</div>
`;

exports[`lib/monitoring/requestpanel/request > should render 1`] = `
<div
  className="msla-panel-about-container"
>
  <div
    className="msla-panel-about-name"
  >
    <Label
      className="msla-panel-connector-label"
      size="large"
    >
      Connector
    </Label>
    <div
      className="msla-panel-connector-name"
    >
      Not available
    </div>
  </div>
  <div
    className="msla-panel-about-description"
  >
    <Label
      className="msla-panel-description-label"
      size="large"
    >
      Operation note
    </Label>
    <div
      className="msla-panel-description"
    >
      <DocumentationItem />
    </div>
  </div>
  <div
    className="msla-panel-about-tags"
  >
    <Label
      className="msla-panel-tags-label"
      size="large"
    >
      Tags
    </Label>
    <div
      className="msla-panel-tags"
    >
      None
    </div>
  </div>
</div>
`;

exports[`lib/monitoring/requestpanel/request > should render a body 1`] = `
<div
  className="msla-panel-about-container"
>
  <div
    className="msla-panel-about-name"
  >
    <Label
      className="msla-panel-connector-label"
      size="large"
    >
      Connector
    </Label>
    <div
      className="msla-panel-connector-name"
    >
      test
    </div>
  </div>
  <div
    className="msla-panel-about-description"
  >
    <Label
      className="msla-panel-description-label"
      size="large"
    >
      Operation note
    </Label>
    <div
      className="msla-panel-description"
    >
      <DocumentationItem
        description="This is a description "
      />
    </div>
  </div>
  <div
    className="msla-panel-about-tags"
  >
    <Label
      className="msla-panel-tags-label"
      size="large"
    >
      Tags
    </Label>
    <div
      className="msla-panel-tags"
    >
      <React.Fragment>
        <div
          aria-label="test"
          className="msla-panel-tag"
          title="Tag1"
        >
          test
        </div>
        <div
          aria-label="more"
          className="msla-panel-tag"
          title="Tag2"
        >
          more
        </div>
      </React.Fragment>
    </div>
  </div>
</div>
`;

exports[`lib/monitoring/requestpanel/request > should render a description with external link 1`] = `
<div
  className="msla-panel-about-container"
>
  <div
    className="msla-panel-about-name"
  >
    <Label
      className="msla-panel-connector-label"
      size="large"
    >
      Connector
    </Label>
    <div
      className="msla-panel-connector-name"
    >
      test
    </div>
  </div>
  <div
    className="msla-panel-about-description"
  >
    <Label
      className="msla-panel-description-label"
      size="large"
    >
      Operation note
    </Label>
    <div
      className="msla-panel-description"
    >
      <DocumentationItem
        description="This is a description "
        link={
          {
            "url": "www.microsoft.com",
            "urlDescription": "Learn more",
          }
        }
      />
    </div>
  </div>
  <div
    className="msla-panel-about-tags"
  >
    <Label
      className="msla-panel-tags-label"
      size="large"
    >
      Tags
    </Label>
    <div
      className="msla-panel-tags"
    >
      <React.Fragment>
        <div
          aria-label="test"
          className="msla-panel-tag"
          title="Tag1"
        >
          test
        </div>
        <div
          aria-label="more"
          className="msla-panel-tag"
          title="Tag2"
        >
          more
        </div>
      </React.Fragment>
    </div>
  </div>
</div>
`;
